### PR TITLE
update tocs with index children

### PR DESCRIPTION
**Pull request by BillWagner**
Update the nodes so the index node is not displayed twice.

These folders had no other changes.

Extracted from #12244

### CODE REVIEW

**Comment by BillWagner** [comment]
@rpetrusha This one is also ready for review.

File: docs/csharp/programming-guide/generics/benefits-of-generics.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Generics provide the solution to a limitation in earlier versions of the common
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/generics/benefits-of-generics.md]: Another nit completely unrelated to your PR: I'm wondering if the **Benefits of generics** topic can be removed. It sounds very much like an attempt to "sell" generics that was written for .NET Framework 2.0. Generics are now an accepted part of the language/framework, so this seems largely to be a statement of the obvious.

File: docs/csharp/programming-guide/statements-expressions-operators/toc.yml [hunk 2]
```diff
@@ -5,38 +5,35 @@
...
-      href: using-conversion-operators.md
-    - name: "How to: Implement User-Defined Conversions Between Structs"
-      href: how-to-implement-user-defined-conversions-between-structs.md
+  - name: Using Conversion Operators
+    href: using-conversion-operators.md
+  - name: "How to: Implement User-Defined Conversions Between Structs"
+    href: how-to-implement-user-defined-conversions-between-structs.md
   - name: default value expressions
     href: default-value-expressions.md
   - name: Equality Comparisons
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/statements-expressions-operators/toc.yml]: I think that I would define a note named "Equality and equality comparisons" and have the three topics (Equality comparisons and the two how to topics) as its children.

File: docs/csharp/programming-guide/statements-expressions-operators/toc.yml [hunk 3]
```diff
@@ -5,38 +5,35 @@
...
   - name: Overloadable Operators
     href: overloadable-operators.md
   - name: Conversion Operators
     href: conversion-operators.md
-    items:
-    - name: Using Conversion Operators
-      href: using-conversion-operators.md
-    - name: "How to: Implement User-Defined Conversions Between Structs"
-      href: how-to-implement-user-defined-conversions-between-structs.md
+  - name: Using Conversion Operators
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/statements-expressions-operators/toc.yml]: Nit: I'm wondering if **Conversion operators** is useful as a parent node, with the three topics as children.

**Review by rpetrusha — APPROVED** [review]
This looks good, @BillWagner. I've left one suggestion, one question, and a couple of comments/nits. You can merge when you're ready.

File: docs/csharp/language-reference/toc.yml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 - name: Language reference
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/toc.yml]: Yes. That's the reason.

Once I've finished this set of PRs, I'm going to make a PR to combine some of the sub-tocs. This is not a good experience.

File: docs/csharp/programming-guide/concepts/async/toc.yml [hunk 5]
```diff
@@ -14,8 +14,9 @@
   - name: Control Flow in Async Programs
     href: control-flow-in-async-programs.md
   - name: Fine-Tuning Your Async Application
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/programming-guide/concepts/async/toc.yml]: I agree, I updated the node.

File: docs/csharp/programming-guide/generics/benefits-of-generics.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/programming-guide/generics/benefits-of-generics.md]: Good suggesetion!  Done.

**Pull request merged**